AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/ReadyEventEmitter.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSPluginsCore
+import Combine
+
+@available(iOS 13.0, *)
+final class ReadyEventEmitter {
+    var readySink: AnyCancellable?
+    init(remoteSyncEnginePublisher: AnyPublisher<RemoteSyncEngineEvent, DataStoreError>) {
+        let queriesReadyPublisher = ReadyEventEmitter.makeSyncQueriesReadyPublisher()
+        let syncEngineStartedPublisher = ReadyEventEmitter.makeRemoteSyncEngineStartedPublisher(
+            remoteSyncEnginePublisher: remoteSyncEnginePublisher
+        )
+        readySink = Publishers
+            .Merge(queriesReadyPublisher, syncEngineStartedPublisher)
+            .sink(receiveCompletion: { completion in
+                switch completion {
+                case .finished:
+                    self.dispatchReady()
+                case .failure(let dataStoreError):
+                    self.log.error("Failed to emit ready event, error: \(dataStoreError)")
```

Comment:
Since this is just an event emitter, the root cause of error is not in this class. The real functioning parts should take care of the error, say if `RemoteSyncEngine` cannot move to state `syncEngineActive`, `RemoteSyncEngine` itself should handle the error.

So, I simply put a `log.error` here.

Reply:
Sounds good.